tests/AwaitTest.php

```diff
@@ -157,6 +157,36 @@ public function testAwaitShouldNotCreateAnyGarbageReferencesForPromiseRejectedWi
         $this->assertEquals(0, gc_collect_cycles());
     }
 
+    /**
+     * @dataProvider provideAwaiters
+     */
+    public function testAlreadyFulfilledPromiseShouldNotSuspendFiber(callable $await)
+    {
+        for ($i = 0; $i < 6; $i++) {
+            $this->assertSame($i, $await(React\Promise\resolve($i)));
+        }
+    }
+
+    /**
+     * @dataProvider provideAwaiters
+     */
+    public function testNestedAwaits(callable $await)
+    {
+        $this->assertTrue($await(new Promise(function ($resolve) use ($await) {
+            $resolve($await(new Promise(function ($resolve) use ($await) {
+                $resolve($await(new Promise(function ($resolve) use ($await) {
+                    $resolve($await(new Promise(function ($resolve) use ($await) {
+                        $resolve($await(new Promise(function ($resolve) use ($await) {
+                            Loop::addTimer(0.01, function () use ($resolve) {
+                                $resolve(true);
+                            });
+                        })));
+                    })));
+                })));
+            })));
+        })));
```

Comment:
What am I looking at? :eyes:

Reply:
:christmas_tree:

Reply:
The idea behind it this is that it can handle mutiple nested awaits on top of each other.